### ISSUE TITLE
Support new submission process for Mahara 23.04

### ISSUE THREAD

**Issue by anitsirk** [catalyst/moodle-assignsubmission_maharaws#54]
For Mahara 23.04, we are changing how submissions are made: Instead of submitting the portfolio itself, a copy is submitted and then managed. This works well for LTI via the 'External Tool' option in Moodle (single page, collection, SmartEvidence collection, portfolio completion collection). However, when I try this with the assignment submission plugin, I get the following error message in Moodle:

```
Attempt to send Mahara request resulted in error: error/webservice call was not successful

[More information about this error](http://docs.moodle.org/400/en/error/assignsubmission_maharaws/errorwsrequest)
Debug info:
Error code: errorwsrequest
Stack trace:

    line 614 of /mod/assign/submission/maharaws/locallib.php: moodle_exception thrown
    line 762 of /mod/assign/submission/maharaws/locallib.php: call to assign_submission_maharaws->submit_view()
    line 7773 of /mod/assign/locallib.php: call to assign_submission_maharaws->save()
    line 7858 of /mod/assign/locallib.php: call to assign->save_submission()
    line 508 of /mod/assign/locallib.php: call to assign->process_save_submission()
    line 55 of /mod/assign/view.php: call to assign->view()

Output buffer: <div class="notifytiny debuggingmessage" data-rel="debugging">Submit view for assessment webservice call failed: 0:error/webservice call was not successful<ul style="text-align: left" data-rel="backtrace"><li>line 613 of /mod/assign/submission/maharaws/locallib.php: call to debugging()</li><li>line 762 of /mod/assign/submission/maharaws/locallib.php: call to assign_submission_maharaws->submit_view()</li><li>line 7773 of /mod/assign/locallib.php: call to assign_submission_maharaws->save()</li><li>line 7858 of /mod/assign/locallib.php: call to assign->save_submission()</li><li>line 508 of /mod/assign/locallib.php: call to assign->process_save_submission()</li><li>line 55 of /mod/assign/view.php: call to assign->view()</li></ul></div>
```

To test:

1. Make sure you have Mahara main (future 23.04).
2. Set up an assignment submission connection.
3. Log in as student and select to submit a portfolio.

Results:
- Expected: A copy is made and that is submitted.
- Actual: The above error message.

Since not everybody will be moving to 23.04 immediately, the old process should still be supported.

For Catalyst staff:

An [assignment](https://webservices-moodle.catalystdemo.co.nz/mod/assign/view.php?id=718) is set up that can be used as basis with test student 'Paula Paulsen'.

**Comment by sarahjcotton** [comment]
Hi @anitsirk - Do you happen to have a timeline for when this might be addressed yet, as I'm about to start work on an upgrade that this will impact.

**Comment by anitsirk** [comment]
@sarahjcotton It is currently being reviewed and the effort and tasks estimated.

**Comment by danmarsden** [comment]
fixed in #58
